src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommandTest.java

```diff
@@ -150,6 +150,36 @@ public void call_successWithSqlScripts() throws IOException, SQLException {
     assertThat(arguments.schemaFilters()).isEmpty();
   }
 
+  @Test
+  public void call_successWithNoNeedQueryText() throws IOException, SQLException {
+    ExtractExecutor executor = Mockito.mock(ExtractExecutor.class);
+    CommandLine cmd = new CommandLine(new ExtractSubcommand(() -> executor, scriptManager));
+
+    assertThat(
+            cmd.execute(
+                "--db-address",
+                "jdbc:hsqldb:mem:my-db1.example",
+                "--db-user",
+                "my-username",
+                "--db-password",
+                "my0password",
+                "--output",
+                outputPath.toString(),
+                "--need-querytext=false"))
+        .isEqualTo(0);
+
+    ArgumentCaptor<ExtractExecutor.Arguments> argumentsCaptor =
+        ArgumentCaptor.forClass(ExtractExecutor.Arguments.class);
+    verify(executor).run(argumentsCaptor.capture());
+    ExtractExecutor.Arguments arguments = argumentsCaptor.getValue();
+
+    assertThat(arguments.outputPath().toString()).isEqualTo(outputPath.toString());
+    assertThat(arguments.sqlScripts()).isEmpty();
+    assertThat(arguments.skipSqlScripts()).isEmpty();
+    assertThat(arguments.chunkRows()).isEqualTo(0);
+    assertThat(arguments.needQueryText()).isFalse();
```

Comment:
In this test only this assert matters and is required. Other asserts may mislead the reader in a way that the `--need-querytext=false` flag somehow affects it. Consider removing

Reply:
It's a bit unfortunate that part of the setup is actually in a separate .sql data file. I agree because of the sheer line numbers needed to define an expected record, it's becoming too DAMP even for the tests.
Changed to keep the most relevant information available without diffusing the readers' attention.

Reply:
Done.